src/main/java/de/tum/cit/aet/artemis/programming/service/BuildLogEntryService.java

```diff
@@ -491,4 +494,51 @@ public boolean buildJobHasLogFile(String buildJobId, ProgrammingExercise program
         return Files.exists(logPath);
     }
 
+    /**
+     * Parses the build log entries from a given file and returns them as a list of {@link BuildLogDTO} objects.
+     *
+     * <p>
+     * The method reads the file line by line and splits each line into a timestamp and a log message.
+     * The timestamp is expected to be separated from the log message by a tab character.
+     * If the timestamp cannot be parsed, the log message is appended to the previous entry.
+     * </p>
+     *
+     * @param buildLog The {@link FileSystemResource} representing the build log file.
+     * @return A list of {@link BuildLogDTO} objects containing the parsed build log entries.
+     */
+    public List<BuildLogDTO> parseBuildLogEntries(FileSystemResource buildLog) {
+        try {
+            List<BuildLogDTO> buildLogEntries = new ArrayList<>();
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(buildLog.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    String[] parts = line.split("\t", 2);
+                    if (parts.length == 2) {
```

Comment:
The 2s seem to be related to me, so I think it would be better to have a local variable for that 
```suggestion
                    // split into timestamp and log message
                    int logMessageParts = 2
                    String[] parts = line.split("\t", logMessageParts);
                    if (parts.length == logMessageParts) {
```

Reply:
Does not need to be changed if there are no other changes